### PR TITLE
chore(cypress): add prettier to cypress

### PR DESCRIPTION
**Pull request by sjohnson-sage**
### Proposed behaviour
- Merge the Cypress eslint config into the root one, and fix all eslint errors in the `./cypress` folder.
- Run prettier over entire `./cypress` folder.
- Change on commit hook to run prettier and eslint over all js files in the repo, rather than just in the `./src` folder.

### Current behaviour
The `./cypress` folder has it's own eslint rules and prettier does not run over it

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: cypress/locators/button-toggle/locators.js [hunk 1]
```diff
@@ -1,2 +1,3 @@
 // component preview locators
-export const BUTTON_TOGGLE_PREVIEW = '[data-component="button-toggle"]';
+const BUTTON_TOGGLE_PREVIEW = '[data-component="button-toggle"]';
+export default BUTTON_TOGGLE_PREVIEW;
```
Comment (DlgSHi) [inline cypress/locators/button-toggle/locators.js:3]: shouldn't be here space?

File: cypress/locators/app-wrapper/locators.js [hunk 2]
```diff
@@ -1,2 +1,3 @@
 // component preview locators
-export const APP_WRAPPER_PREVIEW = '.carbon-app-wrapper';
+const APP_WRAPPER_PREVIEW = ".carbon-app-wrapper";
+export default APP_WRAPPER_PREVIEW;
```
Comment (DlgSHi) [inline cypress/locators/app-wrapper/locators.js:3]: shouldn't there be an empty line?

File: cypress/locators/button-toggle-group/locators.js [hunk 3]
```diff
@@ -1,2 +1,3 @@
 // component preview locators
-export const BUTTON_TOGGLE_GROUP_CONTAINER = '[data-component="button-toggle-group"]';
+const BUTTON_TOGGLE_GROUP_CONTAINER = '[data-component="button-toggle-group"]';
+export default BUTTON_TOGGLE_GROUP_CONTAINER;
```
Comment (DlgSHi) [inline cypress/locators/button-toggle-group/locators.js:3]: add an empty line please

File: cypress/locators/drawer/locators.js [hunk 4]
```diff
@@ -1,2 +1,3 @@
 // component preview locators
-export const DRAWER_TOGGLE = '[data-element="drawer-toggle"]';
+const DRAWER_TOGGLE = '[data-element="drawer-toggle"]';
```
Comment (DlgSHi) [inline cypress/locators/drawer/locators.js:2]: an empty line between 2 and 3 lines?

File: .eslintrc [hunk 5]
```diff
@@ -88,29 +88,34 @@
...
+  "plugins": ["cypress", "react", "react-hooks"],
   "globals": {
     "jest": false,
     "global": false,
     "process": false,
-    "test": false
+    "test": false,
+    "Given": true,
+    "When": true,
+    "Then": true
```
Comment (DlgSHi) [inline .eslintrc:119]: I think we should add also
"And": true

File: cypress/locators/tile/index.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
-import { TILE } from './locators';
+import TILE from "./locators";
 
 // component preview locators
-export const tile = () => cy.get(TILE);
+const tile = () => cy.get(TILE);
+export default tile;
```
Comment (DlgSHi) [inline cypress/locators/tile/index.js:5]: should we add an empty line between 4 and 5 line?

File: cypress/support/step-definitions/configurable-items-steps.js [hunk 7]
```diff
@@ -1,20 +1,30 @@
...
-When('I drag Configurable Items {string} to {int}', (record, destinationId) => {
+When("I drag Configurable Items {string} to {int}", (record, destinationId) => {
   dragAndDrop(draggableItemByText(record), destinationId, startPosition);
 });
 
-When('I drag Configurable Items with iFrame {string} to {int}', (record, destinationId) => {
-  dragAndDrop(draggableItemByTextNoIframe(record), destinationId, startPosition);
-});
+When(
+  "I drag Configurable Items with iFrame {string} to {int}",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/configurable-items-steps.js:15]: it looks not very good IMO, I would prefer to put all the line back and stick with the `When` keyword

File: cypress/support/step-definitions/draggable-context-steps.js [hunk 8]
```diff
@@ -1,11 +1,19 @@
...
-When('I drag Draggable Context {string} to {int}', (record, destinationId) => {
+When("I drag Draggable Context {string} to {int}", (record, destinationId) => {
   const startPosition = 20;
   dragAndDrop(draggableRecordByText(record), destinationId, startPosition);
 });
 
-Then('Draggable Context {string} is dragged to {int}', (record, destinationId) => {
-  draggableRecordByPosition(destinationId).invoke('text').should('contain', record);
-});
+Then(
```
Comment (DlgSHi) [inline cypress/support/step-definitions/draggable-context-steps.js:12]: stick the step name with the `Then` keyword

File: cypress/support/step-definitions/drawer-steps.js [hunk 9]
```diff
@@ -2,45 +2,62 @@ import {
...
-  drawerToggle(drawer).should('have.css', 'transform', 'none');
+Then("toggle {word} Drawers icon switched orientation to closed", (drawer) => {
+  drawerToggle(drawer).should("have.css", "transform", "none");
 });
 
-Then('Drawer {word} animationDuration is set to {string}', (drawer, animationDuration) => {
-  drawerSidebar(drawer).should('have.css', 'animation-duration', animationDuration);
-});
+Then(
+  "Drawer {word} animationDuration is set to {string}",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/drawer-steps.js:55]: the same is here ;)

File: cypress/support/step-definitions/duelling-picklist-steps.js [hunk 10]
```diff
@@ -1,90 +1,108 @@
...
-  duellingPicklistComponent().should('not.have.attr', 'disabled');
+Then("Duelling Picklist is enabled", () => {
+  duellingPicklistComponent().should("not.have.attr", "disabled");
 });
 
-When('I add {int} item(s) from unassigned picklist to assigned picklist', (items) => {
-  for (let i = 0; i < items; i++) {
-    addButton(0).click();
+When(
+  "I add {int} item(s) from unassigned picklist to assigned picklist",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/duelling-picklist-steps.js:59]: and here

File: cypress/support/step-definitions/duelling-picklist-steps.js [hunk 11]
```diff
@@ -1,90 +1,108 @@
...
+  unassignedPicklistItems()
+    .eq(positionOfElement(position))
+    .should("contain", text);
 });
 
-Then('{string} element has golden border outline {string}', (position, color) => {
-  unassignedPicklistItems().eq(positionOfElement(position)).should('have.css', 'outline', color);
-});
+Then(
+  "{string} element has golden border outline {string}",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/duelling-picklist-steps.js:102]: and here

File: cypress/support/step-definitions/flat-table-steps.js [hunk 12]
```diff
@@ -1,173 +1,212 @@
...
+Then("I click on {int} body row", (index) => {
   flatTableBodyRowByPositionInIframe(index).click();
 });
 
-Then('I focus {int} row and focused row element has golden border on focus', (index) => {
-  cy.wait(500, { log: DEBUG_FLAG }); // wait was added due to changing animation
-  flatTableBodyRowByPosition(index).focus().should('have.css', 'outline-color', 'rgb(255, 181, 0)');
-});
+Then(
+  "I focus {int} row and focused row element has golden border on focus",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/flat-table-steps.js:79]: and here

File: cypress/support/step-definitions/grid-steps.js [hunk 13]
```diff
@@ -1,53 +1,59 @@
...
+        `${sizeOfViewport} is not defined in a scope. We want to resize to only for types of viewport`
+      );
   }
 });
 
-Then('pod {int} has height from row {string} to row {string}', (index, rowStart, rowEnd) => {
-  gridPod(index).should('have.css', 'grid-row', `${rowStart} / ${rowEnd}`);
-});
+Then(
+  "pod {int} has height from row {string} to row {string}",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/grid-steps.js:35]: and here

File: cypress/support/step-definitions/grid-steps.js [hunk 14]
```diff
@@ -1,53 +1,59 @@
...
+  (index, rowStart, rowEnd) => {
+    gridPod(index).should("have.css", "grid-row", `${rowStart} / ${rowEnd}`);
+  }
+);
 
-Then('pod {int} has width from column {int} to column {int}', (index, colStart, colEnd) => {
-  gridPod(index).should('have.css', 'grid-column', `${colStart} / ${colEnd}`);
-});
+Then(
+  "pod {int} has width from column {int} to column {int}",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/grid-steps.js:42]: and here

File: cypress/support/step-definitions/grouped-character-steps.js [hunk 15]
```diff
@@ -1,41 +1,57 @@
...
-      case 'third':
-        getKnobsInput(inputFieldName).clear()
-          .then($selector => $selector.val(JSON.stringify($json[2][value]))).type(' ');
-        break;
-      default: throw new Error('There are only three array objects in groupedCharacter.json');
-    }
-  });
-});
+When(
+  "I input json to {string} input field the {string}",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/grouped-character-steps.js:24]: and here

File: cypress/support/step-definitions/i18n-steps.js [hunk 16]
```diff
@@ -1,9 +1,12 @@
...
-  i18nPreview().should('contain', text);
+Then("preview text is {string}", (text) => {
+  i18nPreview().should("contain", text);
 });
 
-Then('preview text is {string} with {string} tag and {string} id', (text, tag, id) => {
-  i18nPreview().children(`${tag}[id=${id}]`).should('have.text', text);
-});
+Then(
+  "preview text is {string} with {string} tag and {string} id",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/i18n-steps.js:8]: and here

File: cypress/support/step-definitions/loader-steps.js [hunk 17]
```diff
@@ -1,34 +1,41 @@
...
-    .and('have.css', 'width', `${width}px`);
-});
-
-Then('Loader isInsideButton and backgroundColor is {string}', (color) => {
-  loaderInsideButton().should('be.visible')
-    .and('have.css', 'background-color', color);
+import { loader, loaderInsideButton } from "../../locators/loader";
+
+Then(
+  "Loader width is set to {int} px and height is set to {int} px",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/loader-steps.js:4]: and here

File: cypress/support/step-definitions/loader-steps.js [hunk 18]
```diff
@@ -1,34 +1,41 @@
...
+  (width, height) => {
+    loader()
+      .should("have.css", "height", `${height}px`)
+      .and("have.css", "width", `${width}px`);
+  }
+);
+
+Then(
+  "button with loader width is set to {int} px and height is set to {int} px",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/loader-steps.js:13]: and here

File: cypress/support/step-definitions/multi-action-button-steps.js [hunk 19]
```diff
@@ -1,44 +1,72 @@
...
 });
 
-Then('Multi Action Button border color is {string} border-color', (borderColor) => {
-  multiActionButton().should('have.css', 'border-bottom-color', borderColor)
-    .and('have.css', 'border-top-color', borderColor)
-    .and('have.css', 'border-right-color', borderColor)
-    .and('have.css', 'border-left-color', borderColor);
-});
+Then(
+  "Multi Action Button border color is {string} border-color",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/multi-action-button-steps.js:18]: and here

File: cypress/support/step-definitions/multi-action-button-steps.js [hunk 20]
```diff
@@ -1,44 +1,72 @@
...
+  multiActionButtonList().eq(2).should("have.text", "Short").and("be.visible");
 });
 
-Then('Multi Action Button in a hidden container is expanded and contains three items', () => {
-  multiActionButtonList().eq(0).should('have.text', 'Button 1').and('be.visible');
-  multiActionButtonList().eq(1).should('have.text', 'Button 2').and('be.visible');
-  multiActionButtonList().eq(2).should('have.text', 'Button 3').and('be.visible');
-});
+Then(
+  "Multi Action Button in a hidden container is expanded and contains three items",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/multi-action-button-steps.js:57]: and here

File: cypress/support/step-definitions/number-input-steps.js [hunk 21]
```diff
@@ -1,58 +1,81 @@
...
-When('I input {int} into NumberInput component', (number) => {
+When("I input {int} into NumberInput component", (number) => {
   commonDataElementInputPreview().clear().type(number);
 });
 
-When('I press keyboard {string} keys into NumberInput input component', (key) => {
-  commonDataElementInputPreview().clear().type(`{${key}}`);
-});
+When(
+  "I press keyboard {string} keys into NumberInput input component",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/number-input-steps.js:26]: and here

File: cypress/support/step-definitions/number-input-steps.js [hunk 22]
```diff
@@ -1,58 +1,81 @@
...
   }
 });
 
-Then('{word} input component size is set to {string} and has min-height set to {int} and paddings set to {int}', (componentName, size, minHeight, px) => {
-  commonDataElementInputPreviewNoIframe().parent().should('have.css', 'min-height', `${minHeight}px`)
-    .and('have.css', 'padding-left', `${px}px`)
-    .and('have.css', 'padding-right', `${px}px`);
-});
+Then(
+  "{word} input component size is set to {string} and has min-height set to {int} and paddings set to {int}",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/number-input-steps.js:66]: and here

File: cypress/support/step-definitions/numeral-date-steps.js [hunk 23]
```diff
@@ -1,50 +1,85 @@
...
+Then("I click on first input", () => {
+  numeralDateInputByPosition(positionOfElement("first")).click();
 });
 
-Then('I type numeral characters {string} in {string} inputs', (text, position) => {
-  numeralDateInputByPosition(positionOfElement(position)).clear().type(text);
-  numeralDateInputByPosition(positionOfElement(position)).parent().should('have.css', 'outline-color', 'rgb(255, 181, 0)');
-});
+Then(
+  "I type numeral characters {string} in {string} inputs",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/numeral-date-steps.js:10]: and here

File: cypress/support/step-definitions/numeral-date-steps.js [hunk 24]
```diff
@@ -1,50 +1,85 @@
...
-When('I leave {string} input empty and blur', (position) => {
+When("I leave {string} input empty and blur", (position) => {
   numeralDateInputByPosition(positionOfElement(position)).clear().blur();
 });
 
-When('I type numeral characters {string} in {string} inputs and blur', (text, position) => {
-  numeralDateInputByPosition(positionOfElement(position)).clear().type(text).blur();
-});
+When(
+  "I type numeral characters {string} in {string} inputs and blur",
```
Comment (DlgSHi) [inline cypress/support/step-definitions/numeral-date-steps.js:24]: and here

File: cypress/locators/button-toggle/locators.js (hunk 1 above)
Comment (sjohnson-sage) [reply cypress/locators/button-toggle/locators.js:3]: Yeah probably

File: cypress/locators/app-wrapper/locators.js (hunk 2 above)
Comment (sjohnson-sage) [reply cypress/locators/app-wrapper/locators.js:3]: There is, it just doesn't show on Github

File: .eslintrc (hunk 5 above)
Comment (sjohnson-sage) [reply .eslintrc:119]: Why? It wasn't there previously, do we need it?

File: cypress/support/step-definitions/configurable-items-steps.js (hunk 7 above)
Comment (sjohnson-sage) [reply cypress/support/step-definitions/configurable-items-steps.js:15]: This is how prettier formats it, prettier is opinionated so it doesn't matter what we individually prefer

**Comment by sjohnson-sage** [comment]
Thanks for reviewing @DlgSHi - we've had this conversation elsewhere but for the benefit of anyone looking at this PR: 

I won't be changing any of the comments for the way prettier has formatted things because the reason we're adding prettier is to have an opinionated formatter making these decisions for us, so that we don't need to have conversations about who prefers which format etc

**Review by DlgSHi — DISMISSED** [review]

**Review by jamime — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by jamime — APPROVED** [review]

**Pull request merged**